### PR TITLE
Smithy and jinja2

### PR DESCRIPTION
**Pull request by kohr-h**
Supersedes #7

### CODE REVIEW

**Comment by kohr-h** [comment]
`make check` fails, not sure why.

**Comment by grlee77** [comment]
not sure what is causing that issue. perhaps try to `cat tests/test-suite.log` and see if that helps?

I can try on an OS X laptop locally tomorrow and see if I can replicate the issue.

**Comment by kohr-h** [comment]
> perhaps try to `cat tests/test-suite.log` and see if that helps?

Oh right, I can do that on Travis. I'll try.

**Comment by grlee77** [comment]
I see the same failure on OSX locally and the contests of tests/test-suite.log indicate there is a failure in linking to FFTW.
```
FAIL: checkall_threads
======================

dyld: Library not loaded: @rpath/libfftw3_threads.3.dylib
  Referenced from: /Users/lee8rx/anaconda/conda-bld/nfft_1537812872358/work/tests/.libs/checkall_threads
  Reason: image not found
FAIL checkall_threads (exit status: 133)
```

I thought the linker flag in build.sh fixed these kind of rpath errors, but apparently it is not working here.

**Comment by grlee77** [comment]
@kohr-h, if you are interested, please add yourself under the maintainers section of `meta.yaml`.

**Comment by kohr-h** [comment]
That's weird.. Does the PR change anything significant in the build environment?

**Comment by kohr-h** [comment]
> @kohr-h, if you are interested, please add yourself under the maintainers section of `meta.yaml`.

Okay sure. In a separate PR or does it suffice to do it here?

**Comment by grlee77** [comment]
> > @kohr-h, if you are interested, please add yourself under the maintainers section of meta.yaml.

> Okay sure. In a separate PR or does it suffice to do it here?

either is fine. I don't mind just adding a commit here.

**Comment by grlee77** [comment]
It seems that it adding export to the LDFLAGS assignment in build.sh fixed the RPATH issue with FFTW.
```
export LDFLAGS="$LDFLAGS -Wl,-rpath,${PREFIX}/lib"
```
However, this just lead to a different error about missing `libgomp.1.dylib`.  That file seems to live within the conda package for `libgfortran`. If I install it as well, then the build succeeds.

However, I think it should be possible to remove `gcc`, go back to the standard compiler approach and add the package `openmp` on OS X which should add openmp support for clang (see conda-forge/clangdev-feedstock#3). I will play with it locally later today when I get a chance and report back.

**Comment by kohr-h** [comment]
Thanks for the investigation @grlee77 I'll give it a try.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,21 +10,22 @@ source:
...
-  number: 0
+  number: 1
   skip: True  # [win]
 
 requirements:
   build:
-    - fftw
-    - gcc
+    - {{ compiler('c') }}
+    - openmp  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Think this needs to be in `host`. Also should be in `run` as well.

xref: https://github.com/conda-forge/python-spams-feedstock/commit/b9801062aa56c9220f6577e32995bc4a4e660b28

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,21 +10,22 @@ source:
...
     - libtool
     - autoconf
     - automake
+  host:
+    - ncurses
```
Comment (jakirkham) [inline recipe/meta.yaml]: Why is `ncurses` needed?

File: recipe/meta.yaml (hunk 1 above)
Comment (kohr-h) [reply recipe/meta.yaml]: Thanks for the heads-up and the reference!

File: recipe/meta.yaml (hunk 2 above)
Comment (kohr-h) [reply recipe/meta.yaml]: Dunno, I'll remove it and see what happens.

Comment (grlee77) [reply recipe/meta.yaml]: The tests seem to link `-lncurses`, but a grep of the source tree didn't find it anywhere else:
https://github.com/NFFT/nfft/blob/7d24b6bb3ec9e185fddae4625403cdbf1f3fa7fa/tests/Makefile.am#L38
So far the CI seems to be passing without it explicitly included.

**Comment by grlee77** [comment]
OS X passed. Thanks, @kohr-h and @jakirkham!

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,21 +10,22 @@ source:
...
     - libtool
     - autoconf
     - automake
+  host:
+    - openmp  # [osx]
+    - fftw
     - cunit
-    - ncurses 5.9*
   run:
+    - openmp  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Would recommend pinning this as in my example.

Comment (kohr-h) [reply recipe/meta.yaml]: Which version would you recommend? Does it matter?

**Comment by grlee77** [comment]
The error on CircleCI was:
```/home/conda/feedstock_root/build_artifacts/nfft_1537836467072/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lncurses```

so it looks like we have to add ncurses back to the list.

**Pull request merged**